app/controllers/admin/categories_controller.rb

```diff
@@ -2,6 +2,9 @@ class Admin::CategoriesController < ApplicationController
 
   before_filter :ensure_is_admin
 
+  #Allow admin to access admin panel before email confirmation
+  skip_filter :cannot_access_without_confirmation
```

Comment:
Is the idea with this change to say that all admin only pages are from now on accessible without email confirmation but all non-admin pages require it? This caught my eye because it appears this skip_filter now goes together with ensure_is_admin filter. skip_filter is a bit of an anti-pattern because it undoes something the parent controller did. Having to do it in a non-exceptional situation (~ in multiple places, as a pattern) indicates that the controller inheritance hierarchy has serious problems.

I have two alternatives in mind here:

* a) introduce a separate admin controller that inherits from application controller and use is as a base class for all admin-only controllers. Similarly introduce a user controller and use it has a base class for non-admin controllers.
* b) remove the filter from application controller and introduce it to all non-admin controllers with before_filter. The option a is preferred because option b makes things only a little bit clearer (explicitly adding something instead of undoing things).

This need for refactoring to admin and user controllers has reared it's head before too. One symptom is these type of skip_filters and other is the sheer size of application controller. We've mitigated the last problem already by moving stuff to helpers and rack middleware. This refactoring would help a lot. The refactoring also doesn't have to be all-in; you can only move the admin and confirmation checks from the application controller to admin controller and user controller and leave rest of the unrelated stuff still to application controller even if it's not truly shared between the two parent controllers. This makes it easier for upcoming changes to keep improving on this by pulling more stuff to these new specialized base controllers.

What do you think?